x-pack/plugins/reporting/server/native_controller_client/synthetic_process.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+import { Readable } from 'stream';
+
+class SyntheticStream extends Readable {
+  constructor(options) {
+    super(options);
+  }
+
+  _read() {
+  }
+}
+
+export class SyntheticProcess {
+
+  _callbacks = {};
+
+  killed = false;
+  stdout = new SyntheticStream();
```

Comment:
I think what you want is a `PassThrough` stream, which you can `write` to and it will just make the written data available to readers. Pretty sure calling `Readable#push()` is not recommended.

Reply:
This was derived from https://nodejs.org/api/stream.html#stream_readable_push_chunk_encoding where they're calling `Readable#push` directly.

Reply:
Does using the PushThrough stream get us any benefits? It appears to be barely documented...

Reply:
from the docs there:

> The readable.push() method is intended be called only by Readable implementers, and only from within the readable._read() method.

`PassThrough` is just a generic Transform stream, so the docs from Readable and Writable apply.